Jenkinsfile

```diff
@@ -183,9 +185,21 @@ pipeline {
                         }
                     }
                 }
+                stage('Mac Unit with Threading') {
+                    agent  { label 'osx' }
+                    steps {
+                        deleteDir()
+                        unstash 'MathSetup'
+                        sh setupCC()
+                        sh "echo CXXFLAGS+=-DSTAN_THREADS >> make/local"
+                        runTests("test/unit")
+                    }
+                    post { always { retry(3) { deleteDir() } } }
+                }
             }
         }
-        stage('Modded tests') {
+        stage('Additional merge tests') {
+            when { anyOf { branch 'develop'; branch 'master' } }
             parallel {
                 stage('Unit with GPU') {
                     agent { label "gelman-group-mac" }
```

Comment:
I think you meant to remove the `agent` label so that it can run either on: "mac / clang 6 libc++ OR linux / gcc 5 stdlibc++ GPU"

Reply:
Thanks for updating the top-level PR comment with this information.